src/components/clusterConfiguration/ClusterConfiguration.tsx

```diff
@@ -143,31 +166,27 @@ const ClusterConfiguration: React.FC<ClusterConfigurationProps> = ({ cluster })
     ingressVip: cluster.ingressVip || '',
     pullSecret: cluster.pullSecret || '',
     sshPublicKey: cluster.sshPublicKey || '',
-    submitType: 'save',
     subnetHostMap: findMatchingSubnet(cluster.ingressVip, cluster.apiVip, subnetHostMap),
   };
 
-  const validationSchema = React.useCallback(
-    () =>
-      Yup.object().shape({
-        name: getUniqueNameValidationSchema(clusterNames).concat(nameValidationSchema),
-        baseDnsDomain: dnsNameValidationSchema,
-        clusterNetworkHostPrefix: hostPrefixValidationSchema,
-        clusterNetworkCidr: ipBlockValidationSchema,
-        serviceNetworkCidr: ipBlockValidationSchema,
-        apiVip: ipValidationSchema,
-        ingressVip: ipValidationSchema,
-        pullSecret: validJSONSchema,
-        sshPublicKey: sshPublicKeyValidationSchema,
-      }),
-    [clusterNames],
-  );
-
-  const handleSubmit = async (values: ClusterConfigurationValues) => {
-    const { submitType, ...params } = values;
+  const handleSubmit = async (
```

Comment:
Should be `React.useCallback()`

Reply:
I'd have to move the whole form into a subcomponent. I've added a TODO note for that.

Reply:
ok, I believe this will not be a performance issue for M1